infinite-tracing/src/test/java/com/newrelic/ResponseObserverTest.java

```diff
@@ -115,5 +118,18 @@ public void shouldTerminateOnALPNError() {
         verify(disconnectionHandler).terminate();
     }
 
-    AtomicBoolean shouldRecreateCall = new AtomicBoolean();
+    @Test
+    public void testIsConnectionTimeoutException() {
+        DisconnectionHandler disconnectionHandler = mock(DisconnectionHandler.class);
+        MetricAggregator metricAggregator = mock(MetricAggregator.class);
+
+        ResponseObserver target = new ResponseObserver(
+                metricAggregator,
+                mock(Logger.class),
+                disconnectionHandler, shouldRecreateCall);
+
+        Throwable exception = new StatusRuntimeException(Status.fromCode(Status.Code.INTERNAL).withDescription("No error: A GRPC status of OK should have been sent\nRst Stream"));
+        assertTrue(target.isConnectionTimeoutException(exception));
```

Comment:
There are a few permutations that are missing from the test suite:
* type is not a `StatusRuntimeException` (tho it might be tested above/below, I didn't check)
* type IS `StatusRuntimeException` but the message doesn't match.
Would be nice to have proper coverage of the new behavior.